pkg/nsx/services/subnetport/subnetport.go

```diff
@@ -286,35 +306,27 @@ func (service *SubnetPortService) ListNSXSubnetPortIDForPod() sets.Set[string] {
 	return subnetPortSet
 }
 
-// TODO: merge the logic to subnet service when subnet implementation is done.
-func (service *SubnetPortService) GetGatewayPrefixForSubnetPort(obj *v1alpha1.SubnetPort, nsxSubnetPath string) (string, int, error) {
-	subnetInfo, err := servicecommon.ParseVPCResourcePath(nsxSubnetPath)
-	if err != nil {
-		return "", -1, err
-	}
-	// TODO: if the port is not the first on the same subnet, try to get the info from existing realized subnetport CR to avoid query NSX API again.
-	statusList, err := service.NSXClient.SubnetStatusClient.List(subnetInfo.OrgID, subnetInfo.ProjectID, subnetInfo.VPCID, subnetInfo.ID)
-	err = nsxutil.TransNSXApiError(err)
-	if err != nil {
-		log.Error(err, "failed to get subnet status")
-		return "", -1, err
+func (service *SubnetPortService) GetGatewayPrefixForSubnetPort(nsxSubnet *model.VpcSubnet) (string, int, error) {
```

Comment:
I'm thinking about, if the precreated Subnet gateway address is updated, and we haven't polled it in our cache yet, here we'll set the old gateway address to SubnetPort.
Would you think this is a acceptable case or do we need to call Subnet status api for precreated Subnet?